bugtool: get bpffs mountpoint from /proc/mounts

jibi opened the pull request:
Rather then hardcoding the /sys/fs/bpf value in bugtool, parse
/proc/mounts to determine the correct mountpoint for the BPF filesystem.

Fixes: #13218

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>

qmonnet commented on bugtool/cmd/configuration.go:
```diff
@@ -39,6 +41,26 @@ func setupDefaultConfig(path string, k8sPods []string, confDir, cmdDir string) (
+	mounts, err := os.Open("/proc/mounts")
+	if err != nil {
+		return ""
+
```
Nit: blank line

qmonnet reviewed (approved):
Looks great, thank you!

This won't be “tested” by the CI, I suppose you validated the change locally to make sure we still dump the maps as intended?

vadorovsky commented on bugtool/cmd/configuration.go:
```diff
@@ -39,6 +41,27 @@ func setupDefaultConfig(path string, k8sPods []string, confDir, cmdDir string) (
 	return &c, save(&c, path)
 }
 
+func bpffsMountpoint() []string {
```
We already have a separate module for listing mounts from `/proc/self/mountinfo` (which contains more details than `/proc/mounts`):

https://github.com/cilium/cilium/blob/master/pkg/mountinfo/mountinfo.go#L125-L135

The example of using it, also for checking BPFFS mount, is here:

https://github.com/cilium/cilium/blob/master/pkg/bpf/bpffs_linux.go#L69-L81

jibi replied:
Thanks @mrostecki, I'll take a look at that package now

vadorovsky reviewed (changes requested):
Thanks for the PR!

We already have a separate package for listing mounts (`pkg/mountinfo`), I think it would be better to use it here.

qmonnet commented on bugtool/cmd/configuration.go:
```diff
@@ -39,6 +40,21 @@ func setupDefaultConfig(path string, k8sPods []string, confDir, cmdDir string) (
+		return ""
+	}
+
+	for _, mountInfo := range mountInfos {
+		if mountInfo.FilesystemType == "bpf" && mountInfo.Root == "/" {
+			return mountInfo.MountPoint
+		}
+	}
```
Can you please add a comment above this block to explain 1) why only list one bpffs and 2) why the additional check on the `"/"` root? Might be helpful to remember the reasoning in the future.

qmonnet reviewed (changes requested):
Looks good to me, please just add a comment so we can remember the logics here.

jibi committed:
bugtool: get bpffs mountpoint from /proc/self/mounts

Rather then hardcoding the /sys/fs/bpf value in bugtool, use the
`mountinfo` package (which exposes the information in /proc/self/mounts)
to determine the correct mountpoint for the BPF filesystem.

Fixes: #13218

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>

qmonnet reviewed (approved):
Looks good to me, thanks!

vadorovsky reviewed (approved):
Looks good. Thanks!
